app/scripts/controllers/app-state.test.js

```diff
@@ -29,5 +44,259 @@ describe('AppStateController', () => {
+      const pollingTokenType = 'popupGasPollTokens';
+      appStateController.addPollingToken('token1', pollingTokenType);
+      expect(appStateController.store.getState()[pollingTokenType]).toContain(
+        'token1',
+      );
+    });
+  });
+
+  describe('removePollingToken', () => {
+    it('removes a pollingToken for a given environmentType', () => {
+      const pollingTokenType = 'popupGasPollTokens';
+      appStateController.addPollingToken('token1', pollingTokenType);
+      appStateController.removePollingToken('token1', pollingTokenType);
+      expect(
+        appStateController.store.getState()[pollingTokenType],
+      ).not.toContain('token1');
+    });
+  });
+
+  describe('clearPollingTokens', () => {
+    it('clears all pollingTokens', () => {
+      appStateController.addPollingToken('token1', 'popupGasPollTokens');
+      appStateController.addPollingToken('token2', 'notificationGasPollTokens');
+      appStateController.addPollingToken('token3', 'fullScreenGasPollTokens');
+      appStateController.clearPollingTokens();
+
+      expect(
+        appStateController.store.getState().popupGasPollTokens,
+      ).toStrictEqual([]);
+      expect(
+        appStateController.store.getState().notificationGasPollTokens,
+      ).toStrictEqual([]);
+      expect(
+        appStateController.store.getState().fullScreenGasPollTokens,
+      ).toStrictEqual([]);
+    });
+  });
+
+  describe('setShowTestnetMessageInDropdown', () => {
+    it('sets whether the testnet dismissal link should be shown in the network dropdown', () => {
+      appStateController.setShowTestnetMessageInDropdown(true);
+      expect(
+        appStateController.store.getState().showTestnetMessageInDropdown,
+      ).toBe(true);
+
+      appStateController.setShowTestnetMessageInDropdown(false);
+      expect(
+        appStateController.store.getState().showTestnetMessageInDropdown,
+      ).toBe(false);
+    });
+  });
+
+  describe('setShowBetaHeader', () => {
+    it('sets whether the beta notification heading on the home page', () => {
+      appStateController.setShowBetaHeader(true);
+      expect(appStateController.store.getState().showBetaHeader).toBe(true);
+
+      appStateController.setShowBetaHeader(false);
+      expect(appStateController.store.getState().showBetaHeader).toBe(false);
+    });
+  });
+
+  describe('setCurrentPopupId', () => {
+    it('sets the currentPopupId in the appState', () => {
+      const popupId = 'popup1';
+
+      appStateController.setCurrentPopupId(popupId);
+      expect(appStateController.store.getState().currentPopupId).toBe(popupId);
+    });
+  });
+
+  describe('getCurrentPopupId', () => {
+    it('retrieves the currentPopupId saved in the appState', () => {
+      const popupId = 'popup1';
+
+      appStateController.setCurrentPopupId(popupId);
+      expect(appStateController.getCurrentPopupId()).toBe(popupId);
+    });
+  });
+
+  describe('setFirstTimeUsedNetwork', () => {
```

Comment:
Just to confirm, have we added all these new unit tests for the general domain for the sake of code coverage?

Reply:
yes, I was hoping those would be enough to reach the target of the codecov.